src/Altinn.Profile/Models/NotificationAddressModel.cs

```diff
@@ -31,32 +31,36 @@ public class NotificationAddressModel : IValidatableObject
         /// <inheritdoc/>
         public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
         {
-            if (string.IsNullOrWhiteSpace(Email) && string.IsNullOrWhiteSpace(Phone))
+            bool hasEmailValue = !string.IsNullOrWhiteSpace(Email);
+            bool hasPhoneValue = !string.IsNullOrWhiteSpace(Phone);
+            bool hasCountryCodeValue = !string.IsNullOrWhiteSpace(CountryCode);
+
+            if (!hasEmailValue && !hasPhoneValue)
             {
                 yield return new ValidationResult("Either Phone or Email must be specified.", [nameof(Phone), nameof(Email)]);
             }
-            else
+            else if (hasEmailValue && hasPhoneValue)
             {
-                if (!string.IsNullOrWhiteSpace(Email) && !string.IsNullOrWhiteSpace(Phone))
-                {
-                    yield return new ValidationResult("Cannot provide both Phone and Email for the same notification address.", [nameof(Phone), nameof(Email)]);
-                }
-
-                if (!string.IsNullOrWhiteSpace(Phone) && string.IsNullOrWhiteSpace(CountryCode))
+                yield return new ValidationResult("Cannot provide both Phone and Email for the same notification address.", [nameof(Phone), nameof(Email)]);
+            }
+            else if (hasPhoneValue)
+            {
+                if (!hasCountryCodeValue)
                 {
                     yield return new ValidationResult("CountryCode is required with Phone.", [nameof(CountryCode)]);
                 }
-
-                if (!string.IsNullOrWhiteSpace(Email) && !string.IsNullOrWhiteSpace(CountryCode))
-                {
-                    yield return new ValidationResult("CountryCode cannot be provided with Email.", [nameof(CountryCode)]);
-                }
-
-                if (string.IsNullOrWhiteSpace(Email) && !IsValidPhoneNumber())
+                else if (!IsValidPhoneNumber())
```

Comment:
I don't think this should be an "else". The idea of a` IEnumerable<ValidationResult>` with `yield return ` is to give all relevant validation errors at the same time. 
```suggestion
                if (!IsValidPhoneNumber())
```

Regarding the statement above (line 42) `if (hasEmailValue && hasPhoneValue)`, it is in the grey zone, where I think both `else if`  and `if `are acceptable solutions

Reply:
I see that Coderabbit disagrees. The validation util for phone number perfectly accepts countrycode + phone when countrycode is null

Reply:
I agree that we should give all relevant validation errors at the same time, and I understand that the validation util doesn't crash if countrycode is null.
I think the key question here is whether or not it is relevant to return "Phone number is not valid" if countryCode is left out.
For me it seems irrelevant (perhaps even misleading) to do a validation on the phone number alone (i.e. when country code is left out), as we need the combination of countrycode and phone number to truly determine the validity of the address. 
An example is a user that wants to add the Swedish number "798765432". `IsValidPhoneNumber` will conclude that this number is invalid without countrycode="+46". But if a request includes the same number combined with the countrycode +46, `IsValidPhoneNumber` is able to see that the number is indeed valid. Do you see my point?

Reply:
Yes, I see your point.